xsuite/src/world/lsworld.ts

```diff
@@ -170,6 +170,47 @@ export class LSWorld extends World {
     return this.proxy.setPreviousBlockInfo(block);
   }
 
+  private async updateCurrentBlockInfo(
```

Comment:
I don't think this abstraction is needed. In `advanceTimestamp`, you could just retrieve the network status and set the increased timestamp, it will make the code much simpler. Maybe later we will want to introduce this abstraction.